restorer_test.go

```diff
@@ -88,38 +87,58 @@ func testRestorerBuilder(buildpackAPI, platformAPI string) func(t *testing.T, wh
 			when("there is no cache", func() {
 				when("there is a cache=true layer", func() {
 					it.Before(func() {
-						var meta string
+						var meta, sha string
 						if api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) < 0 {
 							meta = "cache=true\n"
 						}
-						h.AssertNil(t, writeLayer(layersDir, "buildpack.id", "cache-true", meta, "cache-only-layer-sha"))
+						if api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0 {
+							sha = "cache-only-layer-sha"
+						}
+						h.AssertNil(t, writeLayer(layersDir, "buildpack.id", "cache-true", meta, sha))
 						h.AssertNil(t, restorer.Restore(nil))
 					})
 
-					it("removes metadata and sha file", func() {
+					it("removes metadata file", func() {
 						h.SkipIf(t, api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) >= 0, "Not possible to determine if layers not in cache are launch-only or the result of incorrect metadata")
 
 						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "buildpack.id", "cache-true.toml"))
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "buildpack.id", "cache-true.sha"))
+					})
+
+					it("removes sha file", func() {
+						h.SkipIf(t, api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) >= 0, "Not possible to determine if layers not in cache are launch-only or the result of incorrect metadata")
+						h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) >= 0, "sha file isn't created")
+
+						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "buildpack.id", "cache-true.sha")) // not really needed for platformAPI >= 0.7
 					})
 
 					it("does not restore layer data", func() {
 						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "buildpack.id", "cache-true"))
+						if api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) < 0 {
+							expected := "Removing \"buildpack.id:cache-true\", not in cache"
+							assertLogEntry(t, logHandler, expected)
+						}
 					})
 				})
 
 				when("there is a cache=false layer", func() {
 					it.Before(func() {
-						var meta string
+						var meta, sha string
 						if api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) < 0 {
 							meta = "cache=false"
 						}
-						h.AssertNil(t, writeLayer(layersDir, "buildpack.id", "cache-false", meta, "cache-false-layer-sha"))
+						if api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0 {
+							sha = "cache-false-layer-sha"
```

Comment:
Do you want to add this comment in every place where the sha is an empty string?
I'm wondering if it's necessary because it appears in many places in the tests and I think that it's pretty clear in the `writeLayer` function. Maybe we can add a comment in the `writeLayer` function instead?

Reply:
Ohh fine, let's not duplicate the comment. I wish I didn't have to navigate to the function declaration to understand that an empty sha string results in non-existent sha file, but maybe there isn't a better solution.

Reply:
I added a comment in the function to make it clearer.